code/_onclick/hud/screen_objects/screen_objects.dm

```diff
@@ -671,75 +671,65 @@
 
 	usr.hud_used.hidden_inventory_update()
 
-
+/**
+ * HUD ammo indicator
+ *
+ * Displays a number and an icon representing the ammo for up to 4 at a time
+ */
 /obj/screen/ammo
 	name = "ammo"
 	icon = 'icons/mob/ammoHUD.dmi'
 	icon_state = "ammo"
 	screen_loc = ui_ammo1
+	///If the user has already had their warning played for running out of ammo
 	var/warned = FALSE
+	///Holder for playing a out of ammo animation so that it doesnt get cut during updates
+	var/atom/movable/flash_holder
 	///List of possible screen locs
-	var/static/list/ammo_screen_loc_list = list(ui_ammo1, ui_ammo2, ui_ammo3 ,ui_ammo4)
-
-
-/obj/screen/ammo/proc/add_hud(mob/living/user, obj/item/weapon/gun/G)
-
-	if(isnull(G))
-		CRASH("/obj/screen/ammo/proc/add_hud() has been called from [src] without the required param of G")
-
-	if(!user?.client)
-		return
+	var/static/list/ammo_screen_loc_list = list(ui_ammo1, ui_ammo2, ui_ammo3, ui_ammo4)
 
-	if(!CHECK_BITFIELD(G.flags_gun_features, GUN_AMMO_COUNTER))
-		return
-
-	user.client.screen += src
+/obj/screen/ammo/Initialize()
+	. = ..()
+	flash_holder = new
+	flash_holder.icon_state = "frame"
+	flash_holder.icon = icon
+	flash_holder.plane = plane
+	flash_holder.layer = layer+0.001
+	flash_holder.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	vis_contents += flash_holder
+
+/obj/screen/ammo/Destroy()
+	QDEL_NULL(flash_holder)
+	return ..()
 
+///wrapper to add this to the users screen with a owner
+/obj/screen/ammo/proc/add_hud(mob/living/user, datum/ammo_owner)
+	if(isnull(ammo_owner))
+		CRASH("/obj/screen/ammo/proc/add_hud() has been called from [src] without the required param of ammo_owner")
+	user?.client.screen += src
 
+///wrapper to removing this ammo hud from the users screen
 /obj/screen/ammo/proc/remove_hud(mob/living/user)
 	user?.client?.screen -= src
 
+///actually handles upadating the hud
+/obj/screen/ammo/proc/update_hud(mob/living/user, list/ammo_type, rounds)
+	overlays.Cut()
 
-/obj/screen/ammo/proc/update_hud(mob/living/user, obj/item/weapon/gun/G)
-	if(!user?.client?.screen.Find(src))
-		return
-
-	if(!G || !(G.flags_gun_features & GUN_AMMO_COUNTER))
-		remove_hud(user)
+	if(rounds <= 0)
+		var/hud_state_empty = ammo_type[2]
```

Comment:
Why is that a var ?

Reply:
because "2" isnt very clear, i can make it a define if you prefer